src/datasets/summarizations/trend-regression.summarization.ts

```diff
@@ -0,0 +1,154 @@
+import * as math from 'mathjs';
+import { Summary } from './types';
+import { TimeSeriesPoint } from '../metas/types';
+import { cacheSummaries } from './utils/commons';
+import {
+  MembershipFunction,
+  trapmf,
+  trapmfL,
+  trapmfR,
+  sigmaCountQA,
+} from './libs/protoform';
+import {
+  linearRegression,
+} from './libs/trend';
+import {
+  timeSeriesPointToNumPoint,
+  groupPointsByXWeek,
+} from './utils/time-series';
+import {
+  normalizePoints
+} from './utils/commons';
+
+export function queryFactory(points: TimeSeriesPoint[]) {
+  return cacheSummaries(() => {
+    // The size of the chart is fixed to 800 * 500
+    const MAX_ANGLE = Math.atan(500 / 800);
+    const uQuicklyIncreasingLinearTrend = trapmfL(MAX_ANGLE / 2, MAX_ANGLE * 5 / 8);
+    const uIncreasingLinearTrend = trapmf(MAX_ANGLE / 8, MAX_ANGLE / 4, MAX_ANGLE / 2, MAX_ANGLE * 5 / 8);
+    const uConstantLinearTrend = trapmf(-MAX_ANGLE / 4, -MAX_ANGLE / 8, MAX_ANGLE / 8, MAX_ANGLE / 4);
+    const uDecreasingLinearTrend = trapmf(-MAX_ANGLE * 5 / 8, -MAX_ANGLE / 2, -MAX_ANGLE / 4, -MAX_ANGLE / 8);
+    const uQuicklyDecreasingLinearTrend = trapmfR(-MAX_ANGLE * 5 / 8, -MAX_ANGLE / 2);
+
+    const uSmallRegressionError = trapmfR(0.75, 1.0);
+
+    const uWeekend = (p: TimeSeriesPoint) => {

```

Comment:
probabilityDayIsWeekend better represents that this is a function

Reply:
This is a membership function on time series points. The membership functions in mathematics are usually denoted as "mu" function. That is why I named all the membership functions with "u" as the prefix.

https://en.wikipedia.org/wiki/Membership_function_(mathematics)